sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIOWriteExceptionHandlingTest.java

```diff
@@ -171,9 +163,21 @@ public void testExceptionHandlingForSimpleWrite() throws InterruptedException {
     try {
       pipeline.run().waitUntilFinish();
     } finally {
-      verify(SpannerIO.WriteToSpannerFn.sleeper, times(callsToSleeper)).sleep(anyLong());
-      verify(serviceFactory.mockDatabaseClient(), times(callsToWrite))
-          .writeAtLeastOnceWithOptions(any(), any(Options.ReadQueryUpdateTransactionOption.class));
+      if (exceptionErrorcode == ErrorCode.DEADLINE_EXCEEDED) {
+        // Due to randomOffset retry is done by most likely but not always 9 times.
+        int sleepCalled =
+            Mockito.mockingDetails(SpannerIO.WriteToSpannerFn.sleeper).getInvocations().size();
+        assertTrue(sleepCalled >= 8);
+        assertTrue(sleepCalled <= 10);
```

Comment:
Ran backOff 1 milliion times and found number of retry is distributed as: {8=52661, 9=878266, 10=69073}. i.e. Originally it has a flake probability of 12%

Reply:
That was more effort than I was prepared to do - I was just going to test that steeps >= 8 and  writes >= 9!